dockernet/src/init_chain.sh

```diff
@@ -0,0 +1,178 @@
+    jq '.app_state.gov.deposit_params.max_deposit_period = $newVal' --arg newVal "$MAX_DEPOSIT_PERIOD" $genesis_config > json.tmp && mv json.tmp $genesis_config
+    jq '.app_state.gov.voting_params.voting_period = $newVal' --arg newVal "$VOTING_PERIOD" $genesis_config > json.tmp && mv json.tmp $genesis_config
+    jq '.app_state["tokenfactory"]["params"]["denom_creation_fee"][0]["denom"]=$newVal' --arg newVal "$CREATION_FEE_DENOM" $genesis_config > json.tmp && mv json.tmp $genesis_config
+    jq '.app_state["tokenfactory"]["params"]["denom_creation_fee"][0]["amount"]=$newVal' --arg newVal "$CREATION_FEE_AMOUNT" $genesis_config > json.tmp && mv json.tmp $genesis_config
+}
+
+MAIN_ID=1 # Node responsible for genesis and persistent_peers
+MAIN_NODE_NAME=""
+MAIN_NODE_ID=""
+MAIN_CONFIG=""
+MAIN_GENESIS=""
+echo "Initializing $CHAIN chain..."
+echo "$CHAIN_ID"
+echo "$NUM_NODES"
+for (( i=1; i <= $NUM_NODES; i++ )); do
+    # Node names will be of the form: "simd1"
+    node_name="${NODE_PREFIX}${i}"
+    # Moniker is of the form: SIMD_1
+    moniker=$(printf "${NODE_PREFIX}_${i}" | awk '{ print toupper($0) }')
+
+    # Create a state directory for the current node and initialize the chain
+    mkdir -p $STATE/$node_name
+    
+    # If the chains commands are run only from docker, grab the command from the config
+    # Otherwise, if they're run locally, append the home directory
+    cmd="$BINARY --home ${STATE}/$node_name"
+    # Initialize the chain
+    $cmd init $moniker --chain-id $CHAIN_ID --overwrite &> /dev/null
+    chmod -R 777 $STATE/$node_name
+
+    # Update node networking configuration 
+    config_toml="${STATE}/${node_name}/config/config.toml"
+    client_toml="${STATE}/${node_name}/config/client.toml"
+    app_toml="${STATE}/${node_name}/config/app.toml"
+    genesis_json="${STATE}/${node_name}/config/genesis.json"
+
+    sed -i -E "s|cors_allowed_origins = \[\]|cors_allowed_origins = [\"\*\"]|g" $config_toml
+    sed -i -E "s|127.0.0.1|0.0.0.0|g" $config_toml
+    sed -i -E "s|timeout_commit = \"5s\"|timeout_commit = \"${BLOCK_TIME}\"|g" $config_toml
+    sed -i -E "s|prometheus = false|prometheus = true|g" $config_toml
+
+    sed -i -E "s|minimum-gas-prices = \".*\"|minimum-gas-prices = \"0${DENOM}\"|g" $app_toml
+    sed -i -E '/\[api\]/,/^enable = .*$/ s/^enable = .*$/enable = true/' $app_toml
+    sed -i -E 's|unsafe-cors = .*|unsafe-cors = true|g' $app_toml
+    sed -i -E "s|snapshot-interval = 0|snapshot-interval = 300|g" $app_toml
+    sed -i -E 's|localhost|0.0.0.0|g' $app_toml
+
+    sed -i -E "s|chain-id = \"\"|chain-id = \"${CHAIN_ID}\"|g" $client_toml
+    sed -i -E "s|keyring-backend = \"os\"|keyring-backend = \"test\"|g" $client_toml
+    sed -i -E "s|node = \".*\"|node = \"tcp://localhost:$RPC_PORT\"|g" $client_toml
+
+    sed -i -E "s|\"stake\"|\"${DENOM}\"|g" $genesis_json 
+    sed -i -E "s|\"aphoton\"|\"${DENOM}\"|g" $genesis_json # ethermint default
+
+    # add a validator account
```

Comment:
asking to understand: Why is it that for every node, we're adding val account?

Reply:
Yeah cause each node we need a vaildator.